src/hooks/useAsync.ts

```diff
@@ -33,10 +27,28 @@ const useSafeDispatch = (dispatch: any) => {
     );
 };
 
-const useAsync = (initialState?: any) => {
+export interface UseAsyncResponse<T, U> {
+    data: U extends undefined ? T | null : T;
+    error: any;
+    isError: boolean;
+    isIdle: boolean;
+    isLoading: boolean;
+    isSuccess: boolean;
+    reset: () => any;
+    run: (promise: Promise<any>, dataFetcher?: Function | undefined) => any;
+    setData: (setDataResponse: T) => void;
+    setError: (setErrorResponse: any) => any;
+    status: States;
+}
+
+function useAsync<T>(): UseAsyncResponse<T, undefined>;
+function useAsync<T>(initialData: T): UseAsyncResponse<T, T>;
+function useAsync<T>(initialData?: T): UseAsyncResponse<T, undefined> {
     const initialStateRef = useRef({
         ...defaultInitialState,
-        ...initialState,
+        ...{
+            data: initialData ? initialData : null,
+        },
```

Comment:
Only currently have a use for default the data... so only defaulting that for now. If we later have a reason to default other portions of the state then we can update this then.